.travis.yml

```diff
@@ -8,22 +8,66 @@ services:
 go:
   - 1.5
 
-before_install:
-  - sudo apt-get update
-  - curl https://glide.sh/get | sh
-
-install:
-  - glide install -v
-
-before_script:
-  - docker version
-  - make binary-linux-amd64
-  - chmod +x bin/linux/amd64/local-persist
-  - sudo ./bin/linux/amd64/local-persist &
-
-script:
-  - ./scripts/integration.sh
-  - make test
+jobs:
+  include:
+    - stage: test
+      before_install:
+        - curl https://glide.sh/get | sed 's+get TAG https://glide.sh/version+TAG=v0.12.3+' | sh
+      install:
+        - glide install -v
+      before_script:
+        - make binary-linux-amd64
+        - chmod +x bin/linux/amd64/local-persist
+        - sudo ./bin/linux/amd64/local-persist &
+      script:
+        - ./scripts/integration.sh
+        - make test
+
+    - &deploy-stage
+      stage: build and deploy
+      script:
+        - docker build -t local-persist:$DOCKER_ARCH --build-arg GOARCH=$DOCKER_GOARCH --build-arg GOARM=$DOCKER_GOARM .
+      after_success:
+        - |
+          echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USER" --password-stdin
```

Comment:
Adding my Docker Password to Travis allows anyone to do something in a PR, including outputting my password in plain text. Even if we managed some sort of key, people could publish malicious images with my accounts, etc.

This is something we're probably not gonna have be part of Travis.